x-pack/plugins/osquery/server/routes/fleet_wrapper/get_agents.ts

```diff
@@ -48,6 +48,51 @@ export const getAgentsRoute = (router: IRouter, osqueryContext: OsqueryAppContex
           pitId?: string;
           getStatusSummary?: boolean;
         };
+
+        const internalSavedObjectsClient = await getInternalSavedObjectsClient(
+          osqueryContext.getStartServices
+        );
+        const packagePolicyService = osqueryContext.service.getPackagePolicyService();
+        const agentPolicyService = osqueryContext.service.getAgentPolicyService();
+
+        const { items: packagePolicies } = (await packagePolicyService?.list(
+          internalSavedObjectsClient,
+          {
+            kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.package.name:${OSQUERY_INTEGRATION_NAME}`,
+            perPage: 1000,
+            page: 1,
+          }
+        )) ?? { items: [] as PackagePolicy[] };
+        const supportedPackagePolicyIds = filter(packagePolicies, (packagePolicy) =>
+          satisfies(packagePolicy.package?.version ?? '', '>=0.6.0')
+        );
+        const agentPolicyIds = uniq(map(supportedPackagePolicyIds, 'policy_id'));
+
+        const agentPolicies = await agentPolicyService?.getByIds(
+          internalSavedObjectsClient,
+          agentPolicyIds
+        );
+
+        // FIND agents by policy_name
+        const policyNamePattern = /policy_name:([^ ]+)/;
```

Comment:
since you did not use the `/g` global flag in the pattern, the result of the `match()` function contains the group, too (docs)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#return_value], so you can simply read it from the result array, e.g.

```ts
        const policyName: string | undefined = kuery.match(policyNamePattern)?.[1];

        const foundPolicyByName = policyName
          ? agentPolicies?.filter((policy) =>
              policy.name.toLowerCase().includes(policyName.toLowerCase())
            )
          : [];
```

Reply:
Done! Nice suggestion, thanks! :)